### PR TITLE
fix(internet): filter banned dots from email addresses

### LINKED ISSUE #1882: email address local part shouldnt start or end with .
Inspired by https://github.com/faker-js/faker/discussions/1881

`faker.internet.email("John*","Doe.") //John_Doe.@yahoo.com`

The code is correctly stripping special chars like * but allowing .

BUT email address local parts are not allowed to start or end with a dot or contain multiple consecutive dots

### PR DESCRIPTION
**Pull request by matthewmayer**
fix #1882

### CODE REVIEW

File: src/modules/internet/index.ts [hunk 1]
```diff
@@ -109,6 +109,13 @@ export class InternetModule {
       );
     }
 
+    //local parts may not contain two or more consecutive . characters
+    localPart = localPart.replace(/\.{2,}/g, '.');
+
+    //local parts may not start with or end with a . character
+    localPart = localPart.replace(/^\./, '');
+    localPart = localPart.replace(/\.$/, '');
```
Comment (ST-DDT) [inline src/modules/internet/index.ts:117]: This should also catch multiple dots:

```suggestion
    localPart = localPart.replace(/^\.+/, '');
    localPart = localPart.replace(/\.+$/, '');
```

Comment (ST-DDT) [reply src/modules/internet/index.ts:117]: Alternatively:

```suggestion
    localPart = localPart.replace(/^\.+|\.+$/g, '');
```

File: test/internet.spec.ts [hunk 2]
```diff
@@ -140,6 +140,29 @@ describe('internet', () => {
           expect(faker.definitions.internet.free_email).toContain(suffix);
         });
 
+        it('should not allow an email that starts with a .', () => {
+          const email = faker.internet.email('.Aiden');
```
Comment (ST-DDT) [inline test/internet.spec.ts]:
```suggestion
          const email = faker.internet.email('...Aiden');
```

File: test/internet.spec.ts [hunk 3]
```diff
@@ -140,6 +140,29 @@ describe('internet', () => {
           expect(faker.definitions.internet.free_email).toContain(suffix);
         });
 
+        it('should not allow an email that starts with a .', () => {
```
Comment (ST-DDT) [inline test/internet.spec.ts]: Maybe add a similar test that ends with multiple dots / or combine them into one by having both.

File: src/modules/internet/index.ts (hunk 1 above)
Comment (matthewmayer) [reply src/modules/internet/index.ts:117]: but if we already collapsed multiple dots into one on line 113, there can only be max 1 dot at start or end?

File: test/internet.spec.ts (hunk 3 above)
Comment (matthewmayer) [reply test/internet.spec.ts]: the email function appends numbers etc though, so even if the first name and last name ends with dots, the email normally won't like
`faker.internet.email("Matt...","Mayer....")` // Matt.Mayer.63@gmail.com

File: src/modules/internet/index.ts (hunk 1 above)
Comment (ST-DDT) [reply src/modules/internet/index.ts:117]: Yeah, you are right.

File: src/modules/internet/index.ts [hunk 4]
```diff
@@ -109,6 +109,13 @@ export class InternetModule {
       );
     }
 
+    //local parts may not contain two or more consecutive . characters
```
Comment (Shinigami92) [inline src/modules/internet/index.ts]:
```suggestion
    // local parts may not contain two or more consecutive . characters
```

File: src/modules/internet/index.ts [hunk 5]
```diff
@@ -109,6 +109,13 @@ export class InternetModule {
...
+    localPart = localPart.replace(/\.{2,}/g, '.');
+
+    //local parts may not start with or end with a . character
```
Comment (Shinigami92) [inline src/modules/internet/index.ts]:
```suggestion
    // local parts may not start with or end with a . character
```

**Review by Shinigami92 — DISMISSED** [review]

File: test/internet.spec.ts (hunk 3 above)
Comment (ST-DDT) [reply test/internet.spec.ts]: That may be true now, but what if we add a chance that there isn't an appendix to the name?

I think we can test is by changing the following line to:

````ts
const email = faker.internet.email('...Aiden...', '...Doe...');
````
and then test also for:

````ts
expect(prefix).not.toMatch(/\.{2}/);
expect(prefix).not.toMatch(/\.$/);
````

(and adjust the test name accordingly)

File: test/internet.spec.ts [hunk 6]
```diff
@@ -140,6 +140,29 @@ describe('internet', () => {
...
+          const email = faker.internet.email('...Aiden');
+
+          expect(email).toBeTruthy();
+          expect(email).toBeTypeOf('string');
+          expect(email).toSatisfy(validator.isEmail);
+
+          const [prefix] = email.split('@');
+          expect(prefix).not.toMatch(/^(\.)/);
```
Comment (ST-DDT) [inline test/internet.spec.ts]: Is there a reason why you put the dot in parenthesis?

```suggestion
          expect(prefix).not.toMatch(/^\./);
```

**Review by ST-DDT — CHANGES_REQUESTED** [review]
See above

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
Thank you for fixing this.

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**